h2o-core/src/main/java/water/Paxos.java

```diff
@@ -47,6 +47,12 @@ static synchronized int doHeartbeat( H2ONode h2o ) {
       }
       return 0;
     }
+    
+    if(H2O.isFlatfileEnabled() && h2o._heartbeat._cloud_name_hash != H2O.SELF._heartbeat._cloud_name_hash){
```

Comment:
We can remove it and the method will return earlier in case of node from wrong cluster. I kept the `H2O.isFlatfileEnabled()` there since mutlicast already takes care about this as part of the multicast_hash.

However at the end I would like this change to be valid for both modes. We can remove the `H2O.isFlatfileEnabled()` call, it will work, it's already tested variant.